econml/bootstrap.py

```diff
@@ -5,6 +5,254 @@
+            The variance of the metric of each treatment on each outcome for each sample X[i].
+            Note that when Y or T is a vector rather than a 2-dimensional array,
+            the corresponding singleton dimensions in the output will be collapsed
+            (e.g. if both are vectors, then the output of this method will also be a vector)
+        """
+        return self.stderr**2
+
+    def conf_int(self, alpha=0.1):
+        """
+        Get the confidence interval of the metric of each treatment on each outcome for each sample X[i].
+
+        Parameters
+        ----------
+        alpha: optional float in [0, 1] (Default=0.1)
+            The overall level of confidence of the reported interval.
+            The alpha/2, 1-alpha/2 confidence interval is reported.
+
+        Returns
+        -------
+        lower, upper: tuple of arrays, shape (m, d_y, d_t) or (m, d_y)
+            The lower and the upper bounds of the confidence interval for each quantity.
+            Note that when Y or T is a vector rather than a 2-dimensional array,
+            the corresponding singleton dimensions in the output will be collapsed
+            (e.g. if both are vectors, then the output of this method will also be a vector)
+        """
+        lower = alpha / 2
+        upper = (1 - alpha) / 2
+        if self.kind == 'percentile':
+            return np.percentile(self.pred_dist, lower, axis=0), np.percentile(self.pred_dist, upper, axis=0)
+        elif self.kind == 'pivot':
+            est = self.point_estimate
+            return (2 * est - np.percentile(self.pred_dist, upper, axis=0),
+                    2 * est - np.percentile(self.pred_dist, lower, axis=0))
+        else:
+            raise ValueError("Unrecognized bootstrap kind; valid kinds are 'percentile' and 'pivot'")
+
+    def pvalue(self, value=0):
+        """
+        Get the p value of the each treatment on each outcome for each sample X[i].
+
+        Parameters
+        ----------
+        value: optinal float (default=0)
+            The mean value of the metric you'd like to test under null hypothesis.
+
+        Returns
+        -------
+        pvalue : array-like, shape (m, d_y, d_t) or (m, d_y)
+            The p value of of each treatment on each outcome for each sample X[i].
+            Note that when Y or T is a vector rather than a 2-dimensional array,
+            the corresponding singleton dimensions in the output will be collapsed
+            (e.g. if both are vectors, then the output of this method will also be a vector)
+        """
+
+        if self.kind == 'percentile':
+            dist = self.pred_dist
+        elif self.kind == 'pivot':
+            est = np.mean(self.pred_dist, axis=0)
+            dist = 2 * est - pred_dist
+        else:
+            raise ValueError("Unrecognized bootstrap kind; valid kinds are 'percentile' and 'pivot'")
+        return min((dist < value).sum(), (dist > value).sum()) / dist.shape[0]
```

Comment:
That should only be the case if we choose to make a parametric assumption (that is, if bootstrap_type is 'normal'); otherwise we want to use a non-parametric calculation since the statistic we're bootstrapping may not be normally distributed even asymptotically.